src/App.jsx

```diff
@@ -14,45 +16,86 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SetById = 'id';
+const SetByAlphabet = 'alphabet';
+const SetByLength = 'length';
+const SetByReverse = true;
+const SetByDanger = 'danger';
+
+function getSortedGoods(goods, { sortField, query, reversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SetById:
+        case SetByAlphabet:
+          return good1.localeCompare(good2);
+        case SetByLength:
+          return good1.length - good2.length;
+        default: return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setsortField] = useState(SetByDanger);
+  const [reversed, setReversed] = useState(false);
+  const SortedOptions
```

Comment:
```suggestion
  const sortedGoods
```
usually we name vars in camelCase